dvc/repo/__init__.py

```diff
@@ -264,7 +264,7 @@ def used_cache(
             all_branches=all_branches, all_tags=all_tags
         ):
             if target:
-                if recursive:
+                if recursive and os.path.isdir(target):
```

Comment:
@prihoda could you please briefly explain why do we have this change? Is "used_cached" being used from repro?

Reply:
@shcheklein it's used from push, pull and fetch. I figured it makes sense for all the commands to behave the same way and accept also the stage files.

Reply:
@prihoda yep, makes sense. Thanks!